src/commands/api-mesh/source/install.js

```diff
@@ -0,0 +1,219 @@
+			let [name, version] = source.split('@');
+			const normalizedName = this.normalizeName(name);
+			if (!list[normalizedName]) {
+				this.error(
+					chalk.red(
+						`The source named "${name}" doesn't exist.` +
+							`\nUse "aio api-mesh:source:discover" command to see avaliable sources.`,
+					),
+				);
+			}
+			version = version || list[normalizedName].latest;
+			if (!list[normalizedName].versions.includes(version)) {
+				this.error(
+					chalk.red(
+						`The version "${version}" for source name "${name}" doesn't exist.` +
+							`\nUse "aio api-mesh:source:discover" command to see avaliable source versions.`,
+					),
+				);
+			}
+			const sourceConfig = await this.sourceRegistryStorage.get(name, version);
+			const jsonInterpolate = new JsonInterpolate({ variablesSchema: sourceConfig.variables });
+			const sourceProviderString = JSON.stringify(sourceConfig.provider);
+			const sourceVariables = jsonInterpolate.getJsonVariables(sourceProviderString);
+			const missedVariables = jsonInterpolate.getMissedVariables(variables, sourceVariables);
+			for (const missedVariable of missedVariables) {
+				variables[missedVariable.name] = await promptInput(
+					`Enter the value for variable ${missedVariable.name}:`,
+				);
+			}
+
+			const { error, data } = jsonInterpolate.interpolate(
+				JSON.stringify(sourceConfig.provider),
+				variables,
+			);
+			if (error) {
+				this.error(chalk.red(`${error.message}\n${error.list.map(err => err.message).join('\n')}`));
+			}
+			sourceConfigs.push(JSON.parse(data));
+		}
+
+		try {
+			meshId = await getMeshId(imsOrgId, projectId, workspaceId);
+		} catch (err) {
+			this.error(
+				`Unable to get mesh ID. Please check the details and try again. RequestId: ${global.requestId}`,
+			);
+		}
+
+		if (!meshId) {
+			this.error(
+				`Unable to get mesh config. No mesh found for Org(${imsOrgId}) -> Project(${projectId}) -> Workspace(${workspaceId}). Please check the details and try again.`,
+			);
+		}
+
+		try {
+			const mesh = await getMesh(imsOrgId, projectId, workspaceId, meshId);
+
+			if (!mesh) {
+				this.error(
+					`Unable to get mesh with the ID ${meshId}. Please check the mesh ID and try again. RequestId: ${global.requestId}`,
+					{ exit: false },
+				);
+			}
+			const verifiedSources = this.verifySourceAlreadyExists(
+				mesh.meshConfig.sources,
+				sourceConfigs,
+			);
+			let override = false;
+			if (verifiedSources.installed.length) {
+				override = await promptConfirm(
+					`The following sources are already installed: ${verifiedSources.installed
+						.map(source => source.name)
+						.join(', ')}.
+                    Do you want to override?`,
+				);
+			}
+
+			mesh.meshConfig.sources = override
+				? [...mesh.meshConfig.sources, ...verifiedSources.installed, ...verifiedSources.unique]
+				: [...mesh.meshConfig.sources, ...verifiedSources.unique];
+			try {
+				const response = await updateMesh(imsOrgId, projectId, workspaceId, meshId, {
```

Comment:
You also need to consider the fact that your sources can also have files and they need to be added to the `meshConfig.files`. Probably right now there must be none but its good to cater for it.

Reply:
done